refactor: drop superorder from metas

ojeytonwilliams opened the pull request:
No more needing to remember what the superOrder for a given superBlock is.

ojeytonwilliams committed:
refactor: create superOrder from superblock

ojeytonwilliams committed:
chore: drop superOrders from meta

nhcarrigan commented on curriculum/utils.js:
```diff
@@ -16,3 +16,32 @@ exports.testedLang = function testedLang() {
+    throw Error('superblock must be a string');
+  const order = superBlockToOrder[superblock];
+  if (typeof order === 'undefined')
+    throw Error(`${superblock} is not a valid superblock`);
```
This is throwing on the `certifications` superblock.

ojeytonwilliams replied:
Hmm.  Do we want to give that a `null` `superOrder`?

nhcarrigan replied:
Or filter the test to not care about the `certifications` superblock.

ojeytonwilliams replied:
Could do.  Since `superOrder` doesn't really mean anything for `certifications` probably safest to never set it, though.

ojeytonwilliams committed:
fix: return null for 'certifications'

ojeytonwilliams committed:
fix: only set superOrder if it is needed

nhcarrigan commented:
https://github.com/freeCodeCamp/freeCodeCamp/blob/c98728ccd0619b62ec43bc31ba5c69384063253c/curriculum/test/test-challenges.js#L274

So removing the superOrder effectively disables this test. I removed the filter from this line locally and the tests pass - I can push a commit if you'd like.

ojeytonwilliams commented:
Sure, commit away.

Commit:
fix(tests): remove filter from order tests

ojeytonwilliams committed:
test: use getSuperOrder, since superOrder has gone

ojeytonwilliams committed:
test: ignore certificates

nhcarrigan commented:
@ojeytonwilliams I gave this a rebase to resolve the conflicts with deleting the certification metas.

nhcarrigan commented on curriculum/utils.js:
```diff
@@ -16,3 +16,33 @@ exports.testedLang = function testedLang() {
+  'information-security': 9,
+  'machine-learning-with-python': 10,
+  'coding-interview-prep': 11,
+  certifications: null
```
So I think we can revert all of this logic now that certifications don't have a meta.

ojeytonwilliams replied:
Just drop the 'certifications' special cases? Great, that'll be much nicer.

nhcarrigan replied:
Right because we no longer have certification metas at all, right?

ojeytonwilliams replied:
Sorry, just saw this. Yes, exactly. 

Somehow despite working on both PRs I didn't put 2 and 2 together...

ojeytonwilliams committed:
fix: getSuperOrder throw for 'certifications'

ojeytonwilliams committed:
test: remove certificate special case

It wasn't necessary, anyway (it had never been working because I put
certificate not certification)

moT01 reviewed (approved):
I'm good with this - LGTM :tada:

naomi-lgbt reviewed (approved):
Okay I think I'm happy with this.
